src/main/java/org/broadinstitute/hellbender/engine/filters/flow/WellformedFlowBasedReadFilter.java

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.hellbender.engine.filters.flow;
+
+import htsjdk.samtools.SAMFileHeader;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.engine.filters.AlignmentAgreesWithHeaderReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.engine.filters.WellformedReadFilter;
+import org.broadinstitute.hellbender.utils.help.HelpConstants;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Tests whether a flow based read is &quot;well-formed&quot; -- that is, is free of major internal inconsistencies and issues that could lead
+ * to errors downstream. If a read passes this filter, the rest of the engine should be able to process it without
+ * blowing up. Note that checks already present in WellformedReadFilter are not duplicated here.
+ *
+ * <p><b>Well-formed flow based reads definition</b></p>
+ * <ul>
+ *     <li><b>Flow order: read group must have flow order</b></li>
+ *     <li><b>Quality:</b> should be symtrical within each hmer.</li>
+ *     <li><b>tp attribute:</b> should be symtrical within each hmer.</li>
+ *     <li><b>tp attribute:</b> tp+hmer_length should be within [0, maxhmer],</li>
+ *     <li><b>Hardclipped hmer:</B> is excempted from above checks.</li>
+ *
+ *      @see ReadGroupHasFlowOrderReadFilter
+ *      @see HmerQualitySymetricReadFilter
+ *      @see FlowBasedTPAttributeSymetricReadFilter
+ *      @see FlowBasedTPAttributeValidReadFilter
+ *
+ * </ul>
+ */
+@DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS,
+        groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY,
+        summary = "Keep only flow based reads that are well-formed"
```

Comment:
please re-add the extradocs here like for the standard wellformed readfilter:
```        
extraDocs = {
                ReadFilterLibrary.ValidAlignmentStartReadFilter.class,
                ReadFilterLibrary.ValidAlignmentEndReadFilter.class,
                AlignmentAgreesWithHeaderReadFilter.class,
                ReadFilterLibrary.HasReadGroupReadFilter.class,
                ReadFilterLibrary.MatchingBasesAndQualsReadFilter.class,
                ReadFilterLibrary.ReadLengthEqualsCigarLengthReadFilter.class,
                ReadFilterLibrary.SeqIsStoredReadFilter.class,
                ReadFilterLibrary.CigarContainsNoNOperator.class
                }
                ```

Reply:
Note to self - looks like WellFormedFlowBasedRead is unused. Need to understand

Reply:
@ilyasoifer - this filter does have a test, but it was referencing the name of the filter as a string - therefore there was no reference detected by the IDE. Filters invoked by their class name. Resolved by using the class.getSimpleName() instead of the a string. Resolved